### ISSUE TITLE
Edge case in `clippy::multiple_unsafe_ops_per_block`

### ISSUE THREAD

**Issue by samueltardieu** [rust-lang/rust-clippy#16085]
### Summary

The following example identified by @Jarcho in the discussion of #16079 triggers the lint while it probably shouldn't:

```rust
#![warn(clippy::multiple_unsafe_ops_per_block)]

union U {
    i: u32,
}

#[clippy::msrv = "1.91"]
fn main() {
    let u = U { i: 0 };
    let pu = &raw const u;
    // The lint should probably not trigger here as we can't break apart the expression
    unsafe {
        _ = (*pu).i;
    }
}
```

### Version

```text
Rust nightly-2025-10-31
```
